tools/packaging/packaging.psm1

```diff
@@ -1229,7 +1229,558 @@ function New-ZipPackage
+        '[System.Management.Automation.Internal.CommonParameters.ValidateVariableName]'
+        )
+
+    $reader = [System.IO.File]::OpenText($smaCs)
+    $writer = [System.IO.File]::CreateText($smaCsFiltered)
+
+    while(($line = $reader.ReadLine()) -ne $null)
+    {
+        $match = $line | Select-String -Pattern $patternsToRemove -SimpleMatch
+
+        if($match -ne $null)
+        {
+            $writer.WriteLine("//$line")
+        }
+        else
+        {
+            $writer.WriteLine($line)
+        }
+    }
+    if($reader -ne $null)
+    {
+        $reader.Close()
+    }
+    if($writer -ne $null)
+    {
+        $writer.Close()
+    }
+
+    Move-Item $smaCsFiltered $smaCs -Force
+
+    Write-Verbose -Verbose "Reference assembly code cleanup complete."
+
+    #endregion Cleanup SMA.cs
+
+    #region Build SMA ref assembly
+
+    try
+    {
+        Push-Location .
+        Set-Location $smaProjectFolder
+
+        $csproj =
+@'
+<Project Sdk="Microsoft.NET.Sdk">
+<PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <Version>!replace_version!</Version>
+    <DelaySign>true</DelaySign>
+    <AssemblyOriginatorKeyFile>!replace_snk!</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+</PropertyGroup>
+
+<ItemGroup>
+    <PackageReference Include="Microsoft.Management.Infrastructure" Version="1.0.0-alpha08" />
+    <PackageReference Include="System.Security.AccessControl" Version="4.4.1" />
+    <PackageReference Include="System.Security.Principal.Windows" Version="4.4.1" />
+</ItemGroup>
+</Project>
+'@
+
+        $nugetConfig =
+@'
+<configuration>
+<packageSources>
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
+    <add key="powershell-core" value="https://powershell.myget.org/F/powershell-core/api/v3/index.json" />
+</packageSources>
+</configuration>
+'@
+
+        $csproj = $csproj.Replace('!replace_version!', $RefAssemblyVersion).Replace('!replace_snk!', $SnkFilePath)
+        $csproj | Out-File -FilePath "$smaProjectFolder/System.Management.Automation.csproj" -Force
+
+        $nugetConfig | Out-File -FilePath "$genAPIFolder/Nuget.config" -Force
+
+        dotnet build -c Release
+
+        $refBinPath = Join-Path $smaProjectFolder 'bin/Release/netstandard2.0/System.Management.Automation.dll'
+
+        if($refBinPath -eq $null)
+        {
+            throw "Reference assembly was not built."
+        }
+
+        Copy-Item $refBinPath $RefAssemblyDestinationPath -Force
+
+        Write-Verbose -Verbose "Reference assembly built and copied to $RefAssemblyDestinationPath"
+    }
+    finally
+    {
+        Pop-Location
+    }
+
+    if(Test-Path $genAPIFolder)
+    {
+        Remove-Item $genAPIFolder -Recurse -Force -ErrorAction SilentlyContinue
+    }
+
+    #endregion Build SMA ref assembly
+}
+
 function New-NugetPackage
```

Comment:
Can you please add comments to this function to make it clear that this function is to consume a `.nuspec` and the 
well organized artifacts to create a `.nupkg` file?

Reply:
Fixed.